### PR TITLE
Make an appliance for Python 3

### PR DESCRIPTION
**Pull request by adamnovak**
This is a start on #2742. @DailyDreaming I know it is assigned to you, but I was waiting around for tests to run and I figured I would see if it is actually easy.

### CODE REVIEW

**Comment by adamnovak** [comment]
OK, looks like it is not easy. But maybe this is a starting point? It builds a Python 3 appliance when I `make docker`.

**Comment by adamnovak** [comment]
OK, I think this is only part of #2937, because we still need to actually test the appliance, but it looks like this successfully builds the appliance for Python 3 while calling pip3 to install things, and pushes it to the normal Docker tag, while the Python 2 appliance gets "-py2" appended.

I'm not so sure I like the way I've stuffed things like "what Python are we" into the version file, and I suspect that with multiple minor versions of Python 3 in the world we might want one appliance Docker for each of them, so that might be what review of this should think about.

File: version_template.py [hunk 1]
```diff
@@ -61,23 +64,67 @@ def distVersion():
...
+    workflows to work with a variety of leader Python versions.
+    """
+
+    import sys
+    
+    if sys.version_info[0] == 3:
+        # Ignore minor version
+        return 'python3'
+    else:
+        return exactPython()
```
Comment (DailyDreaming) [inline version_template.py]: Shouldn't we drop py2.7?

**Comment by adamnovak** [comment]
I want to make sure Python 3 is working for everything in master
before we rip out anything we need for Python 2.7 to also work.
Including having manually run a workflow, probably on a Mesos
autoscaling cluster, using the appliance.

**Pull request merged**